binaryfile.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+total=0
+
+for i in $@; do
+   if [ ! -z "${i##[0-9]*}" ]; then
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider more robust numeric validation pattern</b></div>
<div id="fix">

The numeric validation using `${i##[0-9]*}` may not handle negative numbers or decimal values. Consider using a more robust regex pattern or `expr` to validate numeric input.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -6,5 +6,5 @@
 -   if [ ! -z "${i##[0-9]*}" ]; then
 +   if ! [[ $i =~ ^-?[0-9]+(\.[0-9]+)?$ ]]; then
```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Amruta101998/DS-and-Algorithm/pull/26#issuecomment-2514841635>#95e98f</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged